src/factories.js

```diff
@@ -50,21 +46,19 @@ export function createShorthand(Component, mapValueToProps, val, defaultProps =
 
   // Map values to props and create a ReactElement
   if (_.isString(val) || _.isNumber(val)) {
-    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
+    return <Component {...mergePropsAndClassName(mapValueToProps(val), defaultProps)} />
```

Comment:
Would be nice to get some test coverage on these functions. I think that would help make it more clear 

I still don't see why we'd need to switch the order in this case. Using your example:
```js
// Given this definition taken from the Input-v1 PR
const createHTMLInput = createShorthandFactory('input', value => ({ type: value }))

// and this usage within the stardust repo
createHTMLInput(this.props.input, { type: 'password' })

// Usage by a consumer of this lib:
<Input input='text' />

// The shorthand would always yield
<input type='password' />
```

I think about it this way, you can pass props as an object or you can pass a literal that's turned into an object. Both of those cases should be treated the same:
- as an object: `{ type: text }`
- as a literal: `text` is turned into `{type: text}`

I can't how those should be treated differently in terms of overwriting `defaultProps`